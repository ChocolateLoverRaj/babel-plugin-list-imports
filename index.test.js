/* global test, expect */

import createImportLister from './index.js'
import babel from '@babel/core'

test('README.md example', async () => {
  const importLister = createImportLister()

  const code = `
// Normal default and named imports
import React from 'react'
import { niceTool, helper as someHelper } from './some-utils.js'

// Importing for side effects only
import 'side-effects.js'

// Extension doesn't matter
import styles from './styles.css'

// Importing all as namespace
import * as namespaceImport from 'some-lib'

// Exporting from source
export { default as something } from './source.js'

// Exporting everything from a source
export * from './another-source.js'

// Dynamic imports
import('dynamic')

// Dynamic imports referencing an unchanged variable
const dynamicSrc = './tool.js'
var unchangedDynamicSrc = '../path/to/import.js'

import(dynamicSrc)
import(unchangedDynamicSrc)

// Non top-level dynamic imports
document.getElementById('load-button').addEventListener('click', () => {
  import('string-literal')
  import(dynamicSrc)
  import(unchangedDynamicSrc)
})
`
  importLister.once('list', () => {
    expect(importLister.state).toMatchSnapshot()

    importLister.state.clear()
    expect(importLister.state.size).toBe(0)
  })

  babel.transformAsync(code, { plugins: [importLister.plugin] })
})
