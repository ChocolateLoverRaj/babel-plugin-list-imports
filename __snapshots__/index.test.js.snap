// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`README.md example 1`] = `
Set {
  "react",
  "./some-utils.js",
  "side-effects.js",
  "./styles.css",
  "some-lib",
  "./source.js",
  "./another-source.js",
  "dynamic",
  "./tool.js",
  "../path/to/import.js",
  "string-literal",
}
`;
